### ISSUE TITLE
[StructuralMechanicsApplication] question regarding adding properties of NodalConcentratedElement via input json files

### ISSUE THREAD

**Issue by aronnoordam** [KratosMultiphysics/Kratos#11761]
I would like to add the parameters for the NodalConcentratedElement via the projectparameters.json or the materialparameters.json file

If for example I look at a beam element in the StructuralMechanicsApplication. I can add the material parameters in the materialparameters.json. And then in the c++ code, within the element, it retrieves the parameters via `this->GetProperties()`

however, for the NodalConcentratedElement, it retrieves the parameters via e.g.: `this->GetValue(NODAL_MASS)`.  So it is not stored in the properties.

How do i set this "NODAL_MASS" variable as an input in the projectparameters.json or the materialparameters.json

**Comment by loumalouomega** [comment]
I added a process to generate these elements (and more generic ones) in a PR long time ago, here: https://github.com/KratosMultiphysics/Kratos/pull/5627

**Comment by aronnoordam** [comment]
Aha thanks, but as I see the PR is not merged yet, is it currently impossible add parameters to the nodal elements via input files?

**Comment by loumalouomega** [comment]
No, it is not merged and has been blocked for many years, it is a good
moment to push it...

**Comment by philbucher** [comment]
we could never agree on an interface/input format, thats why it is not merged

@aronnoordam do you have a proposal/preference what the UI/UX should be like?

**Comment by loumalouomega** [comment]
> we could never agree on an interface/input format, thats why it is not merged
> 
> @aronnoordam do you have a proposal/preference what the UI/UX should be like?

It is a goood moment to discuss it again!

**Comment by aronnoordam** [comment]
> we could never agree on an interface/input format, thats why it is not merged
> 
> @aronnoordam do you have a proposal/preference what the UI/UX should be like?

For me it would make sense if the manner of input stays consistent with other elements like beams. I.e. that the input parameters are to be defined in the properties in the materialparameters.json. 

But as long as I can define parameters via input files, I would be fine

**Comment by philbucher** [comment]
I think the main issue with putting e.g. the `NODAL_MASS` in the material params is that then it becomes mesh dependent. Perhaps the mesh-dependent settings could be done via a process, and the other properties (like stiffness) in the materials file like for the other elements?

And second question, are you only asking about assigning properties to existing elements, or also create them?

**Comment by aronnoordam** [comment]
imo, it makes more sense to keep all parameters together in the materialparameters.json including NODAL_MASS. Then under the hood, problems regarding mesh dependency can be resolved via a process. I don't see a problem in transfering certain data from the properties to the element under the hood.

Regarding the second question, creating the elements is done via the mdpa file right?

**Comment by loumalouomega** [comment]
In my proposal can be set via properties but also with element-wise values, with Get/SetValue. This way it will be possible to set values for each element in case we want to condesate a system or we have some particular distribution.

**Comment by aronnoordam** [comment]
seems fine by me, what about spring_dampers? Those elements also require nodal parameters. Will this PR also make it possible to input spring_damper parameters via json files?

**Comment by loumalouomega** [comment]
> seems fine by me, what about spring_dampers? Those elements also require nodal parameters. Will this PR also make it possible to input spring_damper parameters via json files?

Do you mean the line ones?, can be added in the same manner if agreed.

**Comment by aronnoordam** [comment]
yes the line ones from: spring_damper_element.cpp

**Comment by aronnoordam** [comment]
@philbucher and @loumalouomega , is it realisticly possible that #5627 and another PR for adding parameters to spring damper elements is merged relatively soon? If not I'll add the possibility outside the StructuralMechanicsApplication, as I need the functionalities

**Comment by loumalouomega** [comment]
I will try to find a moment to adapt #5627 to the spirng damper elements

**Comment by aronnoordam** [comment]
Nice!
